interfaces/apparmor/backend.go

```diff
@@ -0,0 +1,175 @@
+// -*- Mode: Go; indent-tabs-mode: t -*-
+
+/*
+ * Copyright (C) 2016 Canonical Ltd
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License version 3 as
+ * published by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+// Package apparmor implements integration between snappy and
+// ubuntu-core-launcher around apparmor.
+//
+// Snappy creates apparmor profiles for each application (for each snap)
+// present in the system.  Upon each execution of ubuntu-core-launcher, the
+// profile is attached to the running process. Prior to that the profile must
+// be parsed, compiled and loaded into the kernel using the support tool
+// "apparmor_parser".
+//
+// Each apparmor profile contains a simple <header><content><footer> structure.
+// The header specified an identifier that is relevant to the kernel. The
+// identifier can be either the full path of the executable or an abstract
+// identifier not related to the executable name.
+//
+// The actual profiles are stored in /var/lib/snappy/apparmor/profiles.
+// This directory is also hard-coded in ubuntu-core-launcher.
+//
+// NOTE: A systemd job (TODO: specify which) loads all snappy-specific apparmor
+// profiles into the kernel during the boot process.
+package apparmor
+
+import (
+	"bytes"
+	"fmt"
+	"path/filepath"
+
+	"github.com/ubuntu-core/snappy/dirs"
+	"github.com/ubuntu-core/snappy/interfaces"
+	"github.com/ubuntu-core/snappy/osutil"
+	"github.com/ubuntu-core/snappy/snap"
+)
+
+// Backend is responsible for maintaining apparmor profiles for ubuntu-core-launcher.
+type Backend struct {
+	// Custom template exists to support old-security which goes
+	// beyond what is possible with pure security snippets.
+	//
+	// If non-empty then it overrides the built-in template.
+	CustomTemplate string
+}
+
+// Configure creates and loads security artefacts specific to a given snap. The
+// snap can be in developer mode to make security violations non-fatal to the
+// offending application process.
+//
+// This method should be called after changing plug, slots, connections between
+// them or application present in the snap.
+func (b *Backend) Configure(snapInfo *snap.Info, repo *interfaces.Repository, developerMode bool) error {
```

Comment:
Nitpick: the developerMode bool should come right after snapInfo, as it's closely associated with it. This will also fix the inconsistency between Configure and CombineSnippets in that regard.